### PR TITLE
Feature/EPGplayer fallback

### PR DESCRIPTION
**Pull request by aakashRepoCodes**
##  Description
- Handled EPG content playing fallback
1.  Resume the last `channel `from EPG List ( If `channel `exists)
2.  Play the first `program `under the last played `Taxonomy ` (If the `taxonomy` exists)
3.  Play first `program `from the EPG `program `list

## Changes
 -  Resume last watched content on app launch based on fallback logic

##  Test Demo  

- Played `Cooking Channel `, killed app, reopened. The app must play same channel if available in EPG

https://user-images.githubusercontent.com/50236871/227744920-fd689a30-9ec8-469e-ba85-dacb4b3d86f8.mp4

### CODE REVIEW

**Comment by aakashRepoCodes** [comment]
This PR is not ready yet, In progress.

File: app/src/main/java/com/spark/fastplayer/presentation/epg/EPGViewModel.kt [hunk 1]
```diff
@@ -62,6 +68,58 @@ class EPGViewModel @Inject constructor(
             }
             _epgState.value = EPGState.FetchSuccess(map.toList(), set.toList())
         }
+    }
+
+    private fun resumePlaybackFromHistory(epgList: List<EpgRow>) {
+        viewModelScope.launch(coroutineContextProvider.io) {
+            _playbackState.value = PlaybackState.Init
+            if (dataStoreManager.getTaxonomyId.first().isEmpty() || dataStoreManager.getChannelId.first().isEmpty()) {
```
Comment (manish218) [inline app/src/main/java/com/spark/fastplayer/presentation/epg/EPGViewModel.kt:76]: This should be:

if (dataStoreManager.getTaxonomyId.first().isNotEmpty()) {
.....
} else if (dataStoreManager.getTaxonomyId.first().sisNotEmpty()) {
......
} else {
....
}

Comment (manish218) [inline app/src/main/java/com/spark/fastplayer/presentation/epg/EPGViewModel.kt:76]: This should be:
`
if (dataStoreManager.getTaxonomyId.first().isEmpty() && dataStoreManager.getChannelId.first().isEmpty()) {
....
}
`

Comment (aakashRepoCodes) [reply app/src/main/java/com/spark/fastplayer/presentation/epg/EPGViewModel.kt]: done

**Review by manish218 — DISMISSED** [review]

**Review by manish218 — APPROVED** [review]

**Pull request merged**